src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -137,6 +137,9 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                 {
                     await RestoreAsync(request.ForceRestore, request.RestoreSource, token);
                 }
+                catch (OperationCanceledException) when (token.IsCancellationRequested)
```

Comment:
If your pattern is this method returns `false` in cancellation events, I'd count on that being documented in xml doc comments on the method.